File: src/libs/interfaces.go
package libs

// LXCServiceInterface defines the interface for LXC service
type LXCServiceInterface interface {
	Connect() bool
	Disconnect()
	IsConnected() bool
	Execute(command string, timeout *int) (string, *int)
}

// PCTServiceInterface defines the interface for PCT service
type PCTServiceInterface interface {
	Execute(containerID int, command string, timeout *int) (string, *int)
	SetupSSHKey(containerID int, ipAddress string, cfg *LabConfig) bool
	EnsureSSHServiceRunning(containerID int, cfg *LabConfig) bool
	WaitForContainer(containerID int, ipAddress string, cfg *LabConfig, defaultUser string) bool
}

File: src/Libs/Interfaces.cs
namespace Enva.Libs;

public interface ILXCService
{
    bool Connect();
    void Disconnect();
    bool IsConnected();
    (string output, int? exitCode) Execute(string command, int? timeout = null);
}

public interface IPCTService
{
    (string output, int? exitCode) Execute(int containerID, string command, int? timeout = null);
    bool SetupSSHKey(int containerID, string ipAddress, LabConfig cfg);
    bool EnsureSSHServiceRunning(int containerID, LabConfig cfg);
    bool WaitForContainer(int containerID, string ipAddress, LabConfig cfg, string defaultUser);
    ILXCService? GetLXCService();
    (string output, int? exitCode) Start(int containerID);
    (string output, int? exitCode) Stop(int containerID, bool force);
}
